doc/api/process.md

```diff
@@ -637,6 +637,32 @@ process's [`ChildProcess.disconnect()`][].
 If the Node.js process was not spawned with an IPC channel,
 `process.disconnect()` will be `undefined`.
 
+## process.dlopen(module, filename[, flags])
+<!-- YAML
+added: v0.1.16
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/12794
+    description: Added support for the `flags` argument.
+-->
+
+* `module` {Object}
+* `filename` {string}
+* `flags` {integer}.
+
+The `process.dlopen()` method allows to load dynamically shared
+objects. It is primarily used by `require()` to load
+C++ Addons, and should not be used directly expect in special
+cases. One such case is to set dlopen(3) flags.
+
+The `flags` argument is an optional integer that allows to
+specify dlopen behavior. See the [`os.constants.dlopen`][]
+documentation for details.
+
+```js
+process.dlopen(module, require.resolve('binding'));
+```
```

Comment:
A bit of text explaining the use of `require.resolve()` here may be helpful. As it is, this example is a bit terse.
Also, some text about the possible tradeoffs are with bypassing `require()` in this way would be helpful.